### DISCUSSION TITLE
Invalid QR Code

### DISCUSSION THREAD

**Discussion by siim0n** [WalletConnect/walletconnect-monorepo#3298] [Developer Support] [answered]
Hi, I'm trying to setup wallet connect on my dapp, but I'm getting an invalid qr code error from Pera and Defly.
I'm using @txnlab/use-wallet.
My project is not listed on the "explorer" yet, is that the problem ? I didn't really understand that explorer thing, so i'm not sure if it's mandatory.

```ts
import { WalletConnectModalSign } from '@walletconnect/modal-sign-html'
//........
 const walletProviders = useInitializeProviders({
        providers: [
            {
                id: PROVIDER_ID.WALLETCONNECT,
                clientStatic: WalletConnectModalSign,
                clientOptions: {
                    projectId: 'ID',
                    metadata: {
                        name: 'name',
                        description: 'Authenticate yourself',
                        url: 'https://website',
                        icons: ['https://walletconnect.com/walletconnect-logo.png']
                    }
                }
            }
        ]
    })
```

**Comment by glitch-txs** [comment, marked as answer]
Hey, it's possible that these wallets don't support WalletConnect v2 yet, feel free to reach out to them for this matter.
You can see the currently supported wallets in the explorer
https://walletconnect.com/explorer

You don't need to have your project approved to be able to use the ID btw 👌

**Reply by siim0n** [reply]
It seems they don't, I just checked on the website and I don't see either Pera or Defly, thank you :)
